src/server/scripts/Kalimdor/zone_feralas.cpp

```diff
@@ -33,23 +31,42 @@ class npc_gregan_brewspewer : public CreatureScript
     bool OnGossipSelect(Player* player, Creature* creature, uint32 /*sender*/, uint32 action) override
     {
         ClearGossipMenuFor(player);
+
         if (action == GOSSIP_ACTION_INFO_DEF + 1)
         {
-            AddGossipItemFor(player, GOSSIP_ICON_VENDOR, GOSSIP_TEXT_BROWSE_GOODS, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_TRADE);
+            AddGossipItemFor(player, 1801, 0, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 2);
             SendGossipMenuFor(player, 2434, creature->GetGUID());
         }
+
+        if (action == GOSSIP_ACTION_INFO_DEF + 2)
+        {
+            SendGossipMenuFor(player, 2570, creature->GetGUID());
+        }
+
         if (action == GOSSIP_ACTION_TRADE)
+        {
             player->GetSession()->SendListInventory(creature->GetGUID());
+        }
+
         return true;
     }
 
     bool OnGossipHello(Player* player, Creature* creature) override
     {
         if (creature->IsQuestGiver())
+        {
             player->PrepareQuestMenu(creature->GetGUID());
+        }
+
+        if (player->GetQuestStatus(3909) == QUEST_STATUS_INCOMPLETE)
+        {
+            AddGossipItemFor(player, 1802, 0, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+        }
 
-        if (creature->IsVendor() && player->GetQuestStatus(3909) == QUEST_STATUS_INCOMPLETE)
-            AddGossipItemFor(player, GOSSIP_ICON_CHAT, GOSSIP_HELLO, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+        if (creature->IsVendor())
+        {
+            AddGossipItemFor(player, 1802, 1, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_TRADE);
```

Comment:
I'd wrap these magic numbers such as `1802` into defines or constants, for example:

```cpp
#define GOSSIP_TEXT_SOME_DESCRIPTION 1802
```

in order to make them more self-explaining and also avoid repeating the same number twice